src/odemis/driver/pwrmccdaq.py

```diff
@@ -0,0 +1,453 @@
+            old_bit_value = self.device.DBitIn(port, bit)
+            new_bit_value = int(pwr > 0.0)
+            # update the digital output value by using a direct digital port bit
+            if old_bit_value != new_bit_value:
+                logging.debug(f"Setting do_channel {do_ch} from {old_bit_value} to {new_bit_value}")
+                self.device.DBitOut(port, bit, new_bit_value)
+
+
+class MCCDeviceSimulator(object):
+    """
+    A really basic and simple interface to simulate a USB-1208LS device with
+    support for DIO pin read/write commands and a single AO write command.
+    """
+    def __init__(self):
+        # initialize values
+        self.productID = 0x0007a  # USB-1208LS
+        # to keep track of the individual bits and values
+        # self.port_a_bit_config = {"0": 0, "1": 0, "2": 0, "3": 0, "4": 0, "5": 0, "6": 0, "7": 0}
+        # self.port_b_bit_config = {"0": 0, "1": 0, "2": 0, "3": 0, "4": 0, "5": 0, "6": 0, "7": 0}
+        self.port_a_bit_config = [0, 0, 0, 0, 0, 0, 0, 0]
+        self.port_b_bit_config = [0, 0, 0, 0, 0, 0, 0, 0]
+        self.AO_channels = [0, 0]  # value (uint16) in counts to output [10-bits 0-5V]
+
+        # set default configuration
+        self.DConfig(usb_1208LS.DIO_PORTA, 0x00)  # Port A output (all LOW)
+        self.DConfig(usb_1208LS.DIO_PORTB, 0xff)  # Port B input (all HIGH)
+        self.AOut(0, 0x0)
+        self.AOut(1, 0x0)
+
+    def getManufacturer(self):
+        return "MCC"
+
+    def getProduct(self):
+        return "USB-1208LS"
+
+    def getSerialNumber(self):
+        return "021B0CB8"
+
+    def DConfig(self, port_number, bit_mask):
+        """
+        This command sets the direction of the digital bits for a port.
+        :param port_number: AUXPORT = 0x10 | Port A = 0x01 | Port B = 0x04
+        :param bit_mask (int:bit value): 0 = output | 1 = input
+        """
+        if self.productID == 0x0075 and port_number == usb_1208LS.DIO_AUXPORT:
+            bit_mask = ((bit_mask ^ 0xff) & 0xff)
+
+        self.DOut(port_number, bit_mask)
+
+    def DIn(self, port_number):
+        """
+        :param port_number: AUXPORT = 0x10 | Port A = 0x01 | Port B = 0x04
+        :return: the value seen at the port pins
+        """
+        DIO_port = self._return_port_config(port_number)
+        return sum(v << i for i, v in enumerate(DIO_port))
+
+    def DOut(self, port_number, value):
+        """
+        This command writes data to the DIO port bits that are configured as outputs.
+        :param port_number: AUXPORT = 0x10 | Port A = 0x01 | Port B = 0x04
+        :param value: value to write to the port (0-255)
+        """
+        if value < 0 or value > 255:
+            raise ValueError("Value to set is not between 0 and 255")
+
+        DIO_port = self._return_port_config(port_number)
+        in_val = format(value, '#010b')
+        in_val = in_val.replace("0b", "")
+
+        for num, c in enumerate(in_val[::-1]):
+            DIO_port[num] = int(c)
```

Comment:
Please use bit manipulation instead of strings. Also it should only modify the bits that are not masked in Dconfig.

I think in bitwise operation, you would write `dio_port ^=  ((dio_port ^ value) & dio_config)`. But that would require they are all ints.
Here, you can convert `value` to a list of booleans `list_val`, and then do
```
for i in range(8):
    if dio_config[i]
        dio_port[i] = list_val[i]
```